### PR TITLE
Clarify usage of hse_kvs_prefix_delete() in API docs

### PR DESCRIPTION
**Pull request by alexttx**
Fixes https://github.com/hse-project/hse/issues/261

Signed-off-by: Alex Tomlinson <4172734+alexttx@users.noreply.github.com>

### CODE REVIEW

**Review by beckerg — APPROVED** [review]

File: include/hse/hse.h [hunk 1]
```diff
@@ -674,10 +674,9 @@ hse_kvs_param_get(
  * @param flags: Flags for operation specialization.
  * @param txn: Transaction context (optional).
  * @param pfx: Prefix of keys to delete.
- * @param pfx_len: Length of @p pfx.
+ * @param pfx_len: Length of @p pfx, must equal KVS prefix length.
```
Comment (tristan957) [inline include/hse/hse.h:677]: Make the second part a remark

`@remark @p pfx_len must equal the prefix length of @p kvs`

**Review by tristan957 — APPROVED** [review]

File: include/hse/hse.h [hunk 2]
```diff
@@ -653,7 +653,7 @@ hse_kvs_param_get(
  * segmented keys.
  *
  * This interface is used to delete an entire range of segmented keys. To do
- * this the caller passes a filter with a length equal to the KVSs key prefix
+ * this the caller passes a filter with a length equal to the KVS's key prefix
```
[the remark this answers is not in the document]
Comment (alexttx) [reply include/hse/hse.h:656]: We should just find a new initialism to replace KVS!   Key Value Table?

[the remark this answers is not in the document]
Comment (alexttx) [reply include/hse/hse.h:656]: According to MLA, it should be KVS's.

[the remark this answers is not in the document]
Comment (tristan957) [reply include/hse/hse.h:656]: Good to know!

File: include/hse/hse.h (hunk 1 above)
Comment (tristan957) [reply include/hse/hse.h:677]: You didn't respond to this :(

Comment (alexttx) [reply include/hse/hse.h:677]: didn't even see it :(

File: include/hse/hse.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (smoyergh) [reply include/hse/hse.h:656]: Note that we debated this a year ago and landed on "KVSs" and "KVDBs" as the plural forms and updated the docs accordingly in this commit: https://github.com/hse-project/hse-project.github.io/commit/e93b445c82c81c5b89816657b244cb75b6a8bbd1

I see little value in revisiting this decision and having to revise all the docs.

[the remark this answers is not in the document]
Comment (alexttx) [reply include/hse/hse.h:656]: My edit changes this one use to possessive form: "the KVS's key prefix".  There is no plan to change the plural form.

**Pull request merged**